charmcraft/commands/pack.py

```diff
@@ -122,51 +124,54 @@ def run(self, parsed_args: argparse.Namespace) -> None:
         """Run the command."""
         self._check_config(config_file=True)
 
+        config: CharmcraftConfig = self._services.project
+
         builder = package.Builder(
-            config=self.config,
+            config=config,
             force=parsed_args.force,
             debug=parsed_args.debug,
             shell=parsed_args.shell,
             shell_after=parsed_args.shell_after,
             measure=parsed_args.measure,
+            package_service=self._services.package,
         )
 
         # decide if this will work on a charm or a bundle
-        if self.config.type == "charm":
+        if config.type == "charm":
```

Comment:
I suppose the plans are to eviscerate the charm builder at a later point and split its functionality so that the lifecycle run is handled by the lifecycle service, is that correct?